src/bootstrap/src/bin/main.rs

```diff
@@ -33,7 +29,11 @@ fn main() {
         return;
     }
 
+    #[cfg(feature = "tracing")]
+    debug!("parsing flags");
```

Comment:
I would simplify this by creating a logging wrapper macro (`tracing_debug`?) gated from `tracing` feature, which doesn't do anything on `#[cfg(not(feature = "tracing")])` and uses `debug` macro on `#[cfg(feature = "tracing")]`, so you don't need to add this conditional checks everywhere.

Reply:
Good point, I forgor you could just forward via `$($tokens:tt)*`. I'll send a follow-up.

Reply:
Follow-up: #136392